src/fluree/db/query/exec/update.cljc

```diff
@@ -142,22 +142,37 @@
                                  ;;(eg the commit pipeline) relies on
                                  (when-not (iri-mapping? f)
                                    (flake/flip-flake f t))))
+              db-alias   (:alias db)
 
               [s-mch p-mch o-mch] (where/assign-matched-values triple solution nil)
 
-              s-cmp  (if (string? (::where/val s-mch))
-                       (assoc s-mch ::where/val (<? (dbproto/-subid db (::where/val s-mch) true)))
+              s  (or (where/get-sid s-mch db-alias)
+                     (where/get-iri s-mch))
+              p  (or (where/get-sid p-mch db-alias)
+                     (where/get-iri p-mch))
+              o  (or (where/get-value o-mch)
+                     (where/get-iri o-mch)
+                     (where/get-sid o-mch db-alias))
+              dt (::where/datatype o-mch)
+              m  (::where/meta o-mch)
+
+              s-cmp  (if (string? s)
```

Comment:
One of the benefits of separating sids, iris, and values is that you know that `get-iri` should *always* return an iri string, and `get-sid` should always return an sid number, so these `(if (string?)` checks should be redundant. 

There is already an [`evaluate-iris` function](https://github.com/fluree/db/blob/f156d3f8fb729bce6feeee5623dfc77cf4ddf35e/src/fluree/db/query/exec/where.cljc#L380-L397) that ensure the subject, predicate, and object iri matches also have the sids for a given alias. There are also individual functions to evaluate the iris for a subject and predicate that that function uses. 

It looks like this chunk of code is a reimplementation of those functions, and I think we should use them instead. 

If the object here can also be a non-iri value, then we could use those individual functions to evaluate the subject and predicate iris, and use the evaluate-subject-iri function to evaluate the object iri if it's data type is that of an iri.